docs/proposals/api-group.md

```diff
@@ -0,0 +1,104 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/api-group.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Supporting multiple API groups
+
+## Goal
+
+1. Breaking the monolithic v1 API to modular groups
+
+2. Supporting different versions in different groups
+
+3. Supporting identically named kinds to exist in different groups
+
+4. Enabling/disabling groups individually
+
+5. Laying the basis for [API Plugin](../../docs/design/extending-api.md)
+
+6. Allowing users to omit group name when using kubectl
+
+
+## Bookkeeping for groups
+
+1. No changes to API spec:
+
+  We will not add new field in the API spec to hold the group information. Group will be part of the TypeMeta.APIVersion field, in the form of `group/version`.
```

Comment:
I don't think of the two values as being independent.  I would rather have
version and group be linked because as a user I just don't care.

On Thu, Aug 20, 2015 at 2:49 PM, David Eads <notifications@github.com>
wrote:

> In docs/proposals/api-group.md
> <https://github.com/kubernetes/kubernetes/pull/12951#discussion_r37566815>
> :
>
> > +2. Supporting different versions in different groups
> > +
> > +3. Supporting identically named kinds to exist in different groups
> > +
> > +4. Enabling/disabling groups individually
> > +
> > +5. Laying the basis for [API Plugin](../../docs/design/extending-api.md)
> > +
> > +6. Allowing users to omit group name when using kubectl
> > +
> > +
> > +## Bookkeeping for groups
> > +
> > +1. No changes to API spec:
> > +
> > +  We will not add new field in the API spec to hold the group information. Group will be part of the TypeMeta.APIVersion field, in the form of `group/version`.
>
> Why not have a separate field? I don't see the value of forcing two
> distinct values into a single field. I'm fine with the group "" having a
> special meaning for backwards compatibility, but if I have to know the
> correct value anyway, why not separate it out?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/12951/files#r37566815>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift